### ISSUE TITLE
[context] update and connectedCallback ordering causes issues

### ISSUE THREAD

**Issue by trusktr** [webcomponents-cg/community-protocols#63]
I wrote more details here:

- https://github.com/open-wc/context-protocol/issues/27

Basically there is no guarantee that `connectedCallback` of provider and consumer will fire in the expected order. The protocol needs to account for this somehow, in a robust way that makes it easy to follow and for the problem to never appear.

**Comment by Westbrook** [comment]
[This issue](https://github.com/webcomponents-cg/community-protocols/issues/50) references work that became a `context-provider` event from the Lit team that looks to support late registered contexts that may be of use in this situation. Their usage can be found [here](https://github.com/lit/lit/blob/main/packages/context/src/lib/context-root.ts#L59-L87).

**Comment by trusktr** [comment]
Ah, I see. So the concept is basically to add a listener on a node that is known to certainly exist no matter the sub trees, for example document.

Perhaps we should standardize something like `context-provider` in the community protocol.

**Comment by Westbrook** [comment]
Agreed. Maybe we can get @augustjk to share back some of the findings on the Lit side as per #50. If not, it would be great to see an PR with this addition were you to have a little open time, Joe... 😉
